src/app/mat-select-search/mat-select-search.component.ts

```diff
@@ -285,16 +285,51 @@ export class MatSelectSearchComponent implements OnInit, OnDestroy, AfterViewIni
         }
 
         this._options = this.matSelect.options;
+
+
+        // Closure variable for tracking the most recent first option.
+        // In order to avoid avoid causing the list to
+        // scroll to the top when options are added to the bottom of
+        // the list (eg: infinite scroll), we compare only
+        // the changes to the first options to determine if we
+        // should set the first item as active.
+        // This prevents unncessary scrolling to the top of the list
+        // when options are appended, but allows the first item
+        // in the list to be set as active by default when there
+        // is no active selection
+        let previousFirstOption = this._options.toArray()[this.matOption ? 1 : 0];
```

Comment:
how about storing a reference to `option.value` instead of `option`? because the `*ngFor` might recreate the existing options if the reference of the loop item changes when used without `trackBy`

Reply:
I think it is fine like this, since a user can use the `trackBy` function, and distinction by value might not be the best implementation